platform-dependent-components/problem-solver/cxx/generate-response-module/test/units/test.cpp

```diff
@@ -0,0 +1,148 @@
+
+    using AgentTest = ScMemoryTest;
+
+    ScsLoader loader;
+    void loadScsFile(ScMemoryContext & context, std::string const & filename)
+    {
+        loader.loadScsFile(context, TEST_FILES_DIR_PATH + filename);
+    }
+
+    void initialize()
+    {
+        scAgentsCommon::CoreKeynodes::InitGlobal();
+        generateResponseModule::Keynodes::InitGlobal();
+        TestKeynodes::InitGlobal();
+    }
+
+    TEST_F(AgentTest, OneParameterAgentTest)
+    {
+        ScMemoryContext & context = *m_ctx;
+        loadScsFile(context, "baseConcepts.scs");
+        loadScsFile(context, "oneParameterTest.scs");
+
+        ScAgentInit(true);
+        initialize();
+
+        SC_AGENT_REGISTER(OneParameterTestAgent);
+        SC_AGENT_REGISTER(generateResponseModule::GenerateResponseAgent);
+
+        ScAddr const & testActionNode = context.HelperFindBySystemIdtf("test_action_node");
+        ScAddr const & message = context.HelperFindBySystemIdtf("message");
+        ScAddr const & answer = context.HelperFindBySystemIdtf("_answer");
+        
+        utils::AgentUtils::applyAction(&context, testActionNode, WAIT_TIME);
+
+        EXPECT_TRUE(context.HelperCheckEdge(
+                scAgentsCommon::CoreKeynodes::question_finished_successfully,
+                testActionNode,
+                ScType::EdgeAccessConstPosPerm));
+
+        ScAddr messageAnswer = utils::IteratorUtils::getAnyByOutRelation(&context, message, generateResponseModule::Keynodes::nrel_response);
+
+        EXPECT_TRUE(messageAnswer.IsValid());
+
+        ScAddr test_entity = utils::IteratorUtils::getAnyFromSet(&context, messageAnswer);
+
+        EXPECT_TRUE(context.HelperGetSystemIdtf(test_entity) == "test_entity");
+
+        EXPECT_TRUE(context.HelperCheckEdge(answer, messageAnswer, ScType::EdgeAccessConstPosTemp));
+        
+        SC_AGENT_UNREGISTER(OneParameterTestAgent);
+    }
+
+    TEST_F(AgentTest, ZeroParameterTestAgent)
+    {
+        ScMemoryContext & context = *m_ctx;
+        loadScsFile(context, "baseConcepts.scs");
+        loadScsFile(context, "zeroParameterTest.scs");
+
+        ScAgentInit(true);
+        initialize();
+
+        SC_AGENT_REGISTER(ZeroParameterTestAgent);
+        SC_AGENT_REGISTER(generateResponseModule::GenerateResponseAgent);
+
+        ScAddr const & testActionNode = context.HelperFindBySystemIdtf("test_action_node");
+        ScAddr const & message = context.HelperFindBySystemIdtf("message");
+        ScAddr const & answer = context.HelperFindBySystemIdtf("_answer");
+        
+        utils::AgentUtils::applyAction(&context, testActionNode, WAIT_TIME);
+
+        EXPECT_TRUE(context.HelperCheckEdge(
+                scAgentsCommon::CoreKeynodes::question_finished_successfully,
+                testActionNode,
+                ScType::EdgeAccessConstPosPerm));
+
+        ScAddr messageAnswer = utils::IteratorUtils::getAnyByOutRelation(&context, message, generateResponseModule::Keynodes::nrel_response);
```

Comment:
also worth to check that messageAnswer is the same node that being generated by ZeroParameterTestAgent/OneParameterTestAgent (so we can be sure that our response generation agent correctly finds called action answer and uses it)